src/workerd/api/node/node.h

```diff
@@ -17,7 +18,37 @@ namespace workerd::api::node {
 // built-ins
 class CompatibilityFlags : public jsg::Object {
 public:
+
+  bool isEnabled(kj::String flag, workerd::CompatibilityFlags::Reader flags) {
+    KJ_IF_SOME(cached, cachedFlags.find(flag)) {
+      return cached;
+    }
+    auto dynamic  = capnp::toDynamic(flags);
+    auto schema = dynamic.getSchema();
+    for (auto field : schema.getFields()) {
+      for (auto annotation : field.getProto().getAnnotations()) {
+        if (annotation.getId() == COMPAT_ENABLE_FLAG_ANNOTATION_ID) {
+          if (flag == annotation.getValue().getText()) {
+            bool enabled = dynamic.get(field).as<bool>();
+            cachedFlags.upsert(kj::mv(flag), enabled);
+            return enabled;
+          }
+        }
+        else if (annotation.getId() == COMPAT_DISABLE_FLAG_ANNOTATION_ID) {
+          if (flag == annotation.getValue().getText()) {
+            bool enabled = !dynamic.get(field).as<bool>();
+            cachedFlags.upsert(kj::mv(flag), enabled);
+            return enabled;
+          }
+        }
+      }
+    }
+    return false;
```

Comment:
why is defaulting to `false` a good thing? wouldn't that mean that I can easily have a typo in the flag and get a false negative response?

For example: `nodejs_compattt` would yield `false` even when `node_compat` is turned on.

Reply:
Yep. What would you rather it do? My preference would be not to have it throw. Returning `undefined` instead of `false` allows falsy checks to still happen but if someone really needs to differentiate they could.

Reply:
I've reworked it to export constants instead. A flag that doesn't exist will return `undefined` and won't appear as a key in the object.